test/integration/other/empty-entity-structure.js

```diff
@@ -0,0 +1,276 @@
+// an empty structural field like entity.
+
+const emptyEntityForm = `<?xml version="1.0"?>
+<h:html xmlns="http://www.w3.org/2002/xforms" xmlns:h="http://www.w3.org/1999/xhtml" xmlns:jr="http://openrosa.org/javarosa" xmlns:entities="http://www.opendatakit.org/xforms">
+  <h:head>
+    <model entities:entities-version="2023.1.0">
+      <instance>
+        <data id="emptyEntity" orx:version="1.0">
+          <age/>
+          <location>
+            <hometown></hometown>
+          </location>
+          <meta>
+            <entity dataset="people" id="" create="" update="" baseVersion="" />
+          </meta>
+        </data>
+        <other/>
+      </instance>
+      <bind nodeset="/data/age" type="int" entities:saveto="age"/>
+      <bind nodeset="/data/location/hometown" type="string" entities:saveto="hometown"/>
+    </model>
+  </h:head>
+</h:html>`;
+
+const emptyEntitySub = `<data xmlns:jr="http://openrosa.org/javarosa" xmlns:entities="http://www.opendatakit.org/xforms" id="emptyEntity" version="1.0">
+<meta>
+  <instanceID>one</instanceID>
+  <orx:instanceName>one</orx:instanceName>
+  <entity baseVersion="1" dataset="people" id="12345678-1234-4123-8234-123456789abc" update="1"/>
+</meta>
+<age>88</age>
+<location>
+  <hometown></hometown>
+</location>
+</data>`;
+
+describe('empty entity structure field', () => {
+  describe('submission diffing', () => {
+    it('should check simple diff case', testService(async (service) => {
+      const sub2 = emptyEntitySub.replace('<instanceID>one', '<deprecatedID>one</deprecatedID><instanceID>one2')
+        .replace('<hometown></hometown>', '<hometown>seattle</hometown>')
+        .replace('<age>88</age>', '<age></age>');
+
+      const asAlice = await service.login('alice');
+
+      await asAlice.post('/v1/projects/1/forms?publish=true')
+        .send(emptyEntityForm)
+        .set('Content-Type', 'application/xml')
+        .expect(200);
+
+      await asAlice.post('/v1/projects/1/forms/emptyEntity/submissions')
+        .send(emptyEntitySub)
+        .set('Content-Type', 'application/xml')
+        .expect(200);
+
+      await asAlice.put('/v1/projects/1/forms/emptyEntity/submissions/one')
+        .send(sub2)
+        .set('Content-Type', 'application/xml')
+        .expect(200);
+
+      await asAlice.get('/v1/projects/1/forms/emptyEntity/submissions/one/diffs')
+        .set('X-Extended-Metadata', true)
+        .expect(200)
+        .then(({ body }) => {
+          // looks fine without entity child
+          body.one2.should.eql([
+            { old: 'one', new: 'one2', path: [ 'meta', 'instanceID' ] },
+            { new: 'one', path: [ 'meta', 'deprecatedID' ] },
+            { old: '88', new: '', path: [ 'age' ] },
+            { old: '', new: 'seattle', path: [ 'location', 'hometown' ] }
+          ]);
+        });
+    }));
+
+    it('should check diff when form version changes to add label', testService(async (service) => {
```

Comment:
File an issue with this messy diff

<details>
<summary>Example of details block in markdown</summary>

Look, there's cool stuff hidden inside here!
</details>